src/ai/backend/manager/models/group.py

```diff
@@ -116,31 +126,71 @@
 )
 
 
+def _build_group_query(cond, domain_name):
```

Comment:
Let's add type annotation here.
All new codes should have explicit type annotation whenever possible.
(Exception: Recursive data structures such as JSON-based nested dicts could have "abbreviated" type annotations using `Any`.)